src/App.js

```diff
@@ -1,13 +1,124 @@
 import React from 'react';
+import TodoForm from './components/TodoComponents/TodoForm';
+import TodoList from './components/TodoComponents/TodoList';
+import Footer from './components/Footer';
+import complete from './images/complete.svg';
+
+import './App.css';
 
 class App extends React.Component {
-  // you will need a place to store your state in this component.
-  // design `App` to be the parent component of your application.
-  // this component is going to take care of state, and any change handlers you need to work with your state
+  constructor(props) {
+    super(props);
+    this.state = {
+      data: [],
+      value: '',
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.addTodo = this.addTodo.bind(this);
+    this.toggleComplete = this.toggleComplete.bind(this);
+    this.clearCompleted = this.clearCompleted.bind(this);
+  }
+
+  handleChange(e) {
+    this.setState({
+      [e.target.name]: e.target.value,
+    });
+  }
+
+  addTodo(e) {
+    e.preventDefault();
+    const newTodo = {
+      task: this.state.value,
+      id: Date.now(),
+      completed: false,
+    };
+
+    this.setState({ data: [...this.state.data, newTodo], value: '' }, () => {
+      localStorage.setItem('todos', JSON.stringify(this.state.data));
+    });
+  }
```

Comment:
Your way definitely works. We can set local storage without passing setState a callback function. I honestly don't know which way makes more sense.
```
  addTodo(e) {
    e.preventDefault();
    const updatedTodos = [...this.state.data, {
      task: this.state.value,
      id: Date.now(),
      completed: false,
    }];
      localStorage.setItem('todos', JSON.stringify(updatedTodos));
     this.setState({ data: updatedTodos, value: '' });
  }
```